rust/src/importer.rs

```diff
@@ -56,17 +92,29 @@ impl RpmImporter {
         }
     }
 
-    /// Format tmpfiles.d lines for `/var/lib/<foo>` entries.
-    pub fn varlib_tmpfiles_symlinks(&self) -> Vec<String> {
-        let mut tmpfiles_lines = Vec::with_capacity(self.varlib_direntries.len());
+    /// Format tmpfiles.d lines for symlinked entries.
+    // NOTE(lucab): destinations (dirname) can't be quoted as systemd just
+    // parses the remainder of the line, and doesn't expand quotes.
+    pub fn tmpfiles_symlink_entries(&self) -> Vec<String> {
+        let mut tmpfiles_lines =
+            Vec::with_capacity(self.varlib_direntries.len() + self.opt_direntries.len());
+
+        // /opt/ symlinks
+        for dirname in &self.opt_direntries {
+            let linkpath = format!("/opt/{}", dirname);
+            let quoted = crate::maybe_shell_quote(&linkpath);
+            let line = format!("L {quoted} - - - - ../../usr/lib/opt/{dirname}");
+            tmpfiles_lines.push(line);
+        }
```

Comment:
Obviously fine as is, but I think it'd be a bit more elegant to avoid the `Vec::with_capacity` and have these both be mappings, like:

```suggestion
        let direntries = self.opt_direntries.iter().map(|dirname| {
            let linkpath = format!("/opt/{}", dirname);
            let quoted = crate::maybe_shell_quote(&linkpath);
            format!("L {quoted} - - - - ../../usr/lib/opt/{dirname}")
        });

        // same for varlib_direntries
        direntries.chain(varlib_direntries).collect()
```

Then because both implement ExactSizeIterator we should automatically get the right capacity.  (Also, not that it really matters here but with iterators we *could* also use rayon i.e. `.par_iter().map()`)